### PR TITLE
[MRG] Debug convolutional methods that compute barycenters to work with different devices.

### PR DESCRIPTION
**Pull request by framunoz**
## Types of changes

* [X] The `linspace` method of the backends now has the `type_as` argument to convert to the same dtype and device.
* [X] The `convolutional_barycenter2d` and `convolutional_barycenter2d_debiased` functions now work with different devices.

## Motivation and context / Related issue

The problem that the PR solves is that the `convolutional_barycenter2d` and `convolutional_barycenter2d` functions were not working on different devices (e.g. using PyTorch with CUDA). It was detected that the error came from the `linspace` method of the backends not having a `type_as` argument.

## How has this been tested (if it applies)

* Add a test in `test_backend.py` that checks if the `type_as` argument in the `linspace` method works.
* Add the tests `test_wasserstein_bary_2d_d_dtype_device`, `test_wasserstein_bary_2d_device_tf`, `test_wasserstein_bary_2d_debiased_dtype_device` and `test_wasserstein_bary_2d_debiased_device_tf` which mirror the `test_sinkhorn2_variants_dtype_device` and `test_sinkhorn2_variants_device_tf` tests to check if the barycenter work with different dtypes and devices.

## PR checklist

- [X] I have read the [**CONTRIBUTING**](CONTRIBUTING.md) document.
- [X] The documentation is up-to-date with the changes I made (check build artifacts).
- [X] All tests passed, and additional code has been **covered with new tests**.
- [x] I have added the PR and Issue fix to the [**RELEASES.md**](RELEASES.md) file.

### CODE REVIEW

File: test/test_bregman.py [hunk 1]
```diff
@@ -352,13 +352,15 @@ def test_sinkhorn2_variants_device_tf(method):
         nx.assert_same_dtype_device(Mb, Gb)
         nx.assert_same_dtype_device(Mb, lossb)
 
+    # Check that everything happens on the GPU
```
Comment (rflamary) [inline test/test_bregman.py:355]: why did you move this outside of the GPU test?

File: test/test_bregman.py [hunk 2]
```diff
@@ -726,6 +728,137 @@ def test_wasserstein_bary_2d(nx, method):
...
+            nx.assert_same_dtype_device(Ab, bary_wass_b)
+
+
+@pytest.mark.skipif(not tf, reason="tf not installed")
+@pytest.mark.parametrize("method", ["sinkhorn", "sinkhorn_log"])
+def test_wasserstein_bary_2d_device_tf(method):
+    # Using the Tensorflow backend
+    nx = ot.backend.TensorflowBackend()
+
+    rng = np.random.RandomState(42)
```
Comment (rflamary) [inline test/test_bregman.py:786]: could you factorize all this data initialization in a function ad call it each time please (there are other plces where it will be used)?

**Review by rflamary — COMMENTED** [review]
Thanks @framunoz  for you PR, it looks great. Could you answer my question and comment  please?

File: test/test_bregman.py (hunk 1 above)
Comment (framunoz) [reply test/test_bregman.py:355]: Because Codecov does not test these lines, since it works on cpu, giving less coverage to the overall code. When I leave the device verification to the end, Codecov will give coverage to all lines except the last one. Otherwise, it will not give coverage to most of the lines after the `if` as it was before.

To summarize, it is rather a trick to increase the coverage, which were cases that can still be included, but due to limitations of the Codecov machine, it gives the impression that it worsens the coverage.

File: test/test_bregman.py (hunk 2 above)
Comment (framunoz) [reply test/test_bregman.py]: I included a function that initializes the matrix A, and replaced it each time it is called.

**Comment by framunoz** [comment]
The extension that reformats the code to comply with the PEP 8 format made many changes to the code. Do I leave it as it was before the commit? Or is the new style preferred?

**Comment by framunoz** [comment]
I used black in the tests, although because of the vscode extension, the formatting was applied automatically.

In these last commits I left the formatting of the rest of the tests as they were before, also, I added a new line in the `.gitignore` to ignore the `.venv`.
